src/platform/backends/qemu/linux/qemu_platform_detail_linux.cpp

```diff
@@ -192,22 +194,59 @@ QStringList mp::QemuPlatformDetail::vm_platform_args(const VirtualMachineDescrip
          << QString::fromStdString(fmt::format("tap,ifname={},script=no,downscript=no,model=virtio-net-pci,mac={}",
                                                tap_device_name, vm_desc.default_mac_address));
 
+    auto bridge_helper_path = QCoreApplication::applicationDirPath() + "/bridge_helper";
+
+    for (const auto& extra_interface : vm_desc.extra_interfaces)
+    {
+        opts << "-nic"
+             << QString::fromStdString(fmt::format("bridge,br={},model=virtio-net-pci,mac={},helper={}",
+                                                   extra_interface.id,
+                                                   extra_interface.mac_address,
+                                                   bridge_helper_path));
+    }
+
     return opts;
 }
 
-void mp::QemuPlatformDetail::add_network_interface(VirtualMachineDescription& desc,
-                                                   const NetworkInterface& extra_interface)
+std::vector<mp::NetworkInterfaceInfo> mp::QemuPlatformDetail::networks() const
 {
-    // TODO: Do not uncomment the following line when implementing bridging in Linux QEMU. Since implementing it would
-    // yield the same exact implementation than in macOS, we need to move the code out of here, and put it only once
-    // in src/platform/backends/qemu/qemu_virtual_machine.[h|cpp].
-    // desc.extra_interfaces.push_back(net);
+    auto platform_ifs_info = MP_PLATFORM.get_network_interfaces_info();
+
+    std::vector<NetworkInterfaceInfo> networks;
+
+    for (auto& ifs_info : platform_ifs_info)
+    {
+        auto& info = ifs_info.second;
+        const auto& type = info.type;
+
+        if (type == "bridge" || type == "ethernet")
+        {
+            networks.push_back(info);
+        }
+    }
 
-    // For the time being, just complain:
-    throw NotImplementedOnThisBackendException("add interfaces");
+    for (auto& net : networks)
+    {
+        if (net.type == "ethernet")
+        {
+            net.needs_authorization = (MP_BACKEND.find_bridge_with(networks, net.id) == networks.cend());
+        }
+    }
+
+    return networks;
 }
 
 mp::QemuPlatform::UPtr mp::QemuPlatformFactory::make_qemu_platform(const Path& data_dir) const
 {
     return std::make_unique<mp::QemuPlatformDetail>(data_dir);
 }
+
+std::string mp::QemuPlatformDetail::bridge_name_for(const std::string& iface_name) const
+{
+    return MP_BACKEND.bridge_name(iface_name);
```

Comment:
I don't really understand what is going on here. This `bridge_name_for` thing has become pretty complex. This is what I see:

1. the daemon calls `VirtualMachineFactory::bridge_name_for`. This is implemented to:
    1. return an empty string by default. 
        1. this covers libvirt and VirtualBox, which have no specific implementation
    2. In LXD, it returns `mp::Backend::bridge_name()`. 
    3. In QEMU, it delegates on `QemuPlatform::bridge_name_for`, which:
        1. returns an empty string by default (covers macOS, as there is no specific implementation there)
        2. again calls `mp::Backend::bridge_name()` on Linux 
    4. On Hyper-V, it incorrectly returns a hardcoded "External Virtual Switch" string (bug we've discussed)

This complexity is complemented by tests and mocks. But the only functional bit here is `mp::Backend::bridge_name()`, on Linux, where we create bridges. However, and this is the crucial part, IIANM this function shouldn't be used at all in the first place! The daemon should not be looking for the bridge based on the name at all, but based on its members, as we've established when we discussed the bug in Hyper-V. 

Note that information on bridge members is readily available on our return from `VirtualMachineFactory::networks()`. We just need to run a find that is the same for all backends. And this find is implemented. It's the one you just moved [here](https://github.com/canonical/multipass/pull/3463/files#diff-d7e8137b54f1c26de443d2d3364309d322dea72c5ca4299068f7fb23504680b7R270). 

So you could just replace this call [https://github.com/canonical/multipass/blob/main/src/daemon/daemon.cpp#L3523] with a call to `find_bridge_with`. That would fix the behavior and get rid of all this code... Am I missing something??

Reply:
Hey @ricab, thanks! Well, what you are proposing is exactly what I am implementing as the fix to the Hyper-V bug [here](https://github.com/canonical/multipass/pull/3513). What do you think in finishing that fix, merging and then rebasing this branch on top of that? That way, the history will reflect exactly what change does what and this PR will only add bridging to QEMU on Linux.

Otherwise, I can fix everything here. The resulting code will be the same but, you know, the change here can be seen as a drive-by change.

WDYT?

Reply:
Or, I can leave that here, merge this and fix later on the other PR. In any case, both PR's will go to 1.14 IIRC.

Reply:
Ah, I looked at that PR but did not see this. And since this one is adding to the `bridge_name` stuff... But OK, yes, what you propose is perfectly fine! As long as it does the right thing and removes the obsolete code in the end.

Reply:
Of course, I'll let you know when I commit to the other PR. Thanks!

Reply:
Ok, this was all addressed in the other PR. I changed the base of this PR, so now it's on top of the `get` fix.